### PR TITLE
hashicorp vault added

### LINKED ISSUE #66: Hashicorp Vault for secret management

### PR DESCRIPTION
**Pull request by Salem-1**
Please remember to include the new .env, also run entrypoint.sh of vault after building

### CODE REVIEW

**Comment by basheer421** [comment]
Make sure to ignore srcs/backend/backend_tests/email_logs.txt
it's not ignored and it will add conflicts later on.

File: Makefile [hunk 1]
```diff
@@ -20,16 +20,26 @@ help:
 	@echo "  test_frontend   : Run  frontend tests using selenium (make sure node is installed first)"
 	@echo ""
 
-up:
-	docker compose up -d
+unseal:
+	echo please use key to run the project
```
Comment (basheer421) [inline Makefile:24]: add '@' to suppress. "@echo please use key to run the project"

File: Makefile [hunk 2]
```diff
@@ -20,16 +20,26 @@ help:
...
+	@docker exec -it transcendence-vault-1 bash
```
Comment (basheer421) [inline Makefile:25]: Are you sure about using bash for an infinite loop?

**Review by basheer421 — CHANGES_REQUESTED** [review]

**Comment by basheer421** [comment]
The backend is not working, and the website is not loading.
"transcendence-django-1     transcendence-django   "/bin/sh entrypoint.…"   django     7 minutes ago   Restarting (1) 58 seconds ago"


Note: Using Linux incase that matters.

File: Makefile (hunk 2 above)
Comment (Salem-1) [reply Makefile:25]: Bash here is not used for infinite loop, it's used to prompt for unseal keys

**Comment by basheer421** [comment]
make fresh failed
` ERROR [vault 5/5] COPY ./config/vault-config.json /vault/config/vault-config.jso  0.0s`

**Comment by Salem-1** [comment]
> make fresh failed ` ERROR [vault 5/5] COPY ./config/vault-config.json /vault/config/vault-config.jso 0.0s`

I did run the branch locally it worked file, Please delete the branch and pull it again, and let me know the updates

**Comment by Mohammed-Ehsan02** [comment]
backend testing works. Merge with master has no conflicts and works with no issues.

**Pull request merged**